arch/x86/core/intel64/locore.S

```diff
@@ -31,6 +31,17 @@ __start:
 	movw %ax, %fs
 	movw %ax, %gs
 
+	/*
+	 * clear the BSS. note that we do this in 32-bit mode, so
+	 * the BSS must fit entirely in the first 4GB of RAM.
```

Comment:
nitpick: consider adding a build-time check for this somewhere, although this is an astonishingly unlikely situation.